collect_app/src/main/java/org/odk/collect/android/logic/FormDetails.java

```diff
@@ -87,4 +90,48 @@ public boolean isNewerFormVersionAvailable() {
     public boolean areNewerMediaFilesAvailable() {
         return areNewerMediaFilesAvailable;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        FormDetails that = (FormDetails) o;
+        return isNewerFormVersionAvailable() == that.isNewerFormVersionAvailable() &&
```

Comment:
Is all this really necessary? Wouldn't comparing the hash and the error string be sufficient? The download URL would also be needed if somehow you could have downloads from multiple servers going on at the same time but I don't think we have any intention of making that possible. It seems weird to compare these status objects, anyway. Where is this used? Or is it just hashing that ends up getting used?

Reply:
This is just using the Android Studio equals/hash generator. Probably good to add a comment when I do that. Generally I'd argue for "data" objects like this it makes sense to just compare the whole thing (like we'd get with Lombok or Kotlin) unless we have a very good reason to do otherwise.

I generated this for testing with mockito but I ended up converting to a fake (which makes life easier) so I can just kill this code.